horace_core/sqw/@sqw/get_efix.m

```diff
@@ -29,111 +29,24 @@
 
 % Original author: T.G.Perring
 %
-% $Revision:: 1759 ($Date:: 2020-02-10 16:06:00 +0000 (Mon, 10 Feb 2020) $)
-
 
 % Parse input
 % -----------
-[w, args, mess] = horace_function_parse_input (nargout,varargin{:});
-if ~isempty(mess), error(mess); end
-
-
-% Perform operations
-% ------------------
-tol_default=5e-3;
-if numel(args)==1 && isnumeric(args{1}) && isscalar(args{1}) && args{1}>=0
-    tol=args{1};
-elseif numel(args)==0
-    tol=tol_default;    % relative tolerance of spread of incident energies
-else
-    error('Check optional fractional tolerance is a non-negative scalar')
-end
-
-% Check that the data has the correct type
-if ~all(w.sqw_type(:))
-    error('efix and emode can only be retrived from sqw-type data')
-end
-
-source_is_file=w.source_is_file;
-nobj=numel(w.data);     % number of sqw objects or files
-nfiles=w.nfiles;
-nend=cumsum(nfiles(:));
-nbeg=nend-nfiles(:)+1;
-efix_arr=zeros(nend(end),1);
-emode_arr=zeros(nend(end),1);
-for i=1:nobj
-    if source_is_file
-        ld = w.loaders_list{i};
-        header = ld.get_header('-all');
-    else
-        header=w.data(i).experiment_info;
-    end
-    [efix_arr(nbeg(i):nend(i)),emode_arr(nbeg(i):nend(i))]=get_efix_single(header);
-end
-
-en=struct('efix',efix_arr,'emode',emode_arr,'ave',NaN,'min',NaN,'max',NaN,'relerr',NaN);
-if all(emode_arr==emode_arr(1))
-    efix=sum(efix_arr)/numel(efix_arr);
-    emode=emode_arr(1);
-    en.ave=efix;
-    en.min=min(efix_arr);
-    en.max=max(efix_arr);
-    if en.ave==0 && en.min==0 && en.max==0
-        en.relerr=0;    % if all energies==0, then accept this as no relative error
-    else
-        en.relerr=max(en.max-efix,efix-en.min)./efix;
-    end
-    if isfinite(en.relerr) && abs(en.relerr)<=tol
-        ok=true;
-        mess='';
-    else
-        ok=false;
-        mess=['Spread of efix lies outside acceptable fraction of average of ',num2str(tol)];
+if exist('tol','var') 
+    if ~(isnumeric(tol) && isscalar(tol) && tol >=0)
+        error( ...
+            'Check optional fractional tolerance is a non-negative scalar')
```

Comment:
```suggestion
    validateattributes(tol, {'numeric'}, {'scalar', 'nonnegative'})
```

Reply:
did not like this due to confusing errors it generates